plugins/catalog-backend-module-gcp/src/providers/GkeEntityProvider.ts

```diff
@@ -0,0 +1,193 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  TaskRunner,
+  readTaskScheduleDefinitionFromConfig,
+} from '@backstage/backend-tasks';
+import {
+  DeferredEntity,
+  EntityProvider,
+  EntityProviderConnection,
+} from '@backstage/plugin-catalog-node';
+
+import { Logger } from 'winston';
+import * as container from '@google-cloud/container';
+import {
+  ANNOTATION_KUBERNETES_API_SERVER,
+  ANNOTATION_KUBERNETES_API_SERVER_CA,
+  ANNOTATION_KUBERNETES_AUTH_PROVIDER,
+} from '@backstage/plugin-kubernetes-common';
+import { Config } from '@backstage/config';
+import { SchedulerService } from '@backstage/backend-plugin-api';
+
+/**
+ * Catalog provider to ingest GKE clusters
+ *
+ * @public
+ */
+export class GkeEntityProvider implements EntityProvider {
+  private readonly logger: Logger;
+  private readonly scheduleFn: () => Promise<void>;
+  private readonly gkeParents: string[];
+  private readonly clusterManagerClient: container.v1.ClusterManagerClient;
+  private connection?: EntityProviderConnection;
+
+  constructor(
+    logger: Logger,
+    taskRunner: TaskRunner,
+    gkeParents: string[],
+    clusterManagerClient: container.v1.ClusterManagerClient,
+  ) {
+    this.logger = logger;
+    this.scheduleFn = this.createScheduleFn(taskRunner);
+    this.gkeParents = gkeParents;
+    this.clusterManagerClient = clusterManagerClient;
+  }
+
+  public static fromConfig(
+    logger: Logger,
+    scheduler: SchedulerService,
+    config: Config,
+  ) {
+    const gkeProviderConfig = config.getConfig('catalog.providers.gcp.gke');
+    const schedule = readTaskScheduleDefinitionFromConfig(
+      gkeProviderConfig.getConfig('schedule'),
+    );
+    return new GkeEntityProvider(
+      logger,
+      scheduler.createScheduledTaskRunner(schedule),
+      gkeProviderConfig.getStringArray('parents'),
+      new container.v1.ClusterManagerClient(),
+    );
+  }
+
+  getProviderName(): string {
+    return `gcp-gke`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    await this.scheduleFn();
+  }
+
+  private filterOutUndefinedDeferredEntity(
+    e: DeferredEntity | undefined,
+  ): e is DeferredEntity {
+    return e !== undefined;
+  }
+
+  private filterOutUndefinedCluster(
+    c: container.protos.google.container.v1.ICluster | null | undefined,
+  ): c is container.protos.google.container.v1.ICluster {
+    return c !== undefined && c !== null;
+  }
+
+  private clusterToResource(
+    cluster: container.protos.google.container.v1.ICluster,
+  ): DeferredEntity | undefined {
+    const location = cluster.location;
```

Comment:
Not 100% sure what to use as location

Reply:
Think it's fine. You can have something like `${this.getProviderName()}:${url}` or something?